Source/InstrumentEditorVRC7.cpp

```diff
@@ -101,10 +146,71 @@ BOOL CInstrumentEditorVRC7::OnInitDialog()
 		_T("Synthesizer"),
 		_T("Chorus")
 	};
+	const _TCHAR* const YM2413_PATCH_NAME[16] = {
+		_T("(custom patch)"),
+		_T("Violin"),
+		_T("Guitar"),
+		_T("Piano"),
+		_T("Flute"),
+		_T("Clarinet"),
+		_T("Oboe"),
+		_T("Trumpet"),
+		_T("Organ"),
+		_T("Horn"),
+		_T("Synthesizer"),
+		_T("Harpsichord"),
+		_T("Vibraphone"),
+		_T("Synthsizer Bass"),
+		_T("Acoustic Bass"),
+		_T("Electric Guitar")
+	};
+	const _TCHAR* const YMF281B_PATCH_NAME[16] = {
+		_T("(custom patch)"),
+		_T("Electric Strings"),
+		_T("Bow Wow"),
+		_T("Electric Guitar"),
+		_T("Organ"),
+		_T("Clarinet"),
+		_T("Saxophone"),
+		_T("Trumpet"),
+		_T("Street Organ"),
+		_T("Synth Brass"),
+		_T("Electric Piano"),
+		_T("Bass"),
+		_T("Vibraphone"),
+		_T("Chimes"),
+		_T("Tom Tom II"),
+		_T("Noise")
+	};
 
-	for (int i = 0; i < 16; ++i) {
-		Text.Format(_T("Patch #%i - %s"), i, PATCH_NAME[i]);
-		pPatchBox->AddString(Text);
+	switch (m_iPatchTone) {
+	// various VRC7 patch versions
+	case VRC7_NUKE:
+	case VRC7_RW:
+	case VRC7_FT36:
+	case VRC7_FT35:
+	case VRC7_MO:
+	case VRC7_KT2:
+	case VRC7_KT1:
+		for (int i = 0; i < 16; ++i) {
+			Text.Format(_T("Patch #%i - %s"), i, VRC7_PATCH_NAME[i]);
+			pPatchBox->AddString(Text);
+		}
+		break;
+	// YM2413 tone by Mitsutaka Okazaki, 2020
+	case TONE_2413:
+		for (int i = 0; i < 16; ++i) {
+			Text.Format(_T("Patch #%i - %s"), i, YM2413_PATCH_NAME[i]);
+			pPatchBox->AddString(Text);
+		}
+		break;
+	// YMF281B tone by Chabin (4/10/2004) with fixes from plgDavid
+	case TONE_281B:
+		for (int i = 0; i < 16; ++i) {
+			Text.Format(_T("Patch #%i - %s"), i, YMF281B_PATCH_NAME[i]);
+			pPatchBox->AddString(Text);
+		}
+		break;
 	}
```

Comment:
can you deduplicate this code? declare `const _TCHAR* patch_names;` assign it as one of the arrays depending on which patch bank is in use, then print it once at the end.